File: 0328.odd_even_linked_list/solution.go
package leetcode0328

import "github.com/Ackerr/Algorithms/utils"

func oddEvenList(head *utils.ListNode) *utils.ListNode {
	if head == nil {
		return head
	}
	odd, even := head, head.Next
	evenHead := even
	for even != nil && even.Next != nil {
		odd.Next = even.Next
		even.Next = even.Next.Next
		odd = odd.Next
		even = even.Next
	}
	odd.Next = evenHead
	return head
}

File: 0328.odd_even_linked_list/solution.py
from utils import ListNode


class Solution:
    def oddEvenList(self, head: ListNode) -> ListNode:
        if not head:
            return head
        old, even = head, head.next
        even_head = head.next
        even = even
        while even and even.next:
            old.next = even.next
            even.next = even.next.next
            old = old.next
            even = even.next
        old.next = even_head
        return head
